app/controllers/uploads_controller.rb

```diff
@@ -53,9 +45,30 @@ def show
 
   private
 
+  def alert_messages(loaded_csv)
+    total_rows_count = loaded_csv.ids.length
+    failed_rows = loaded_csv.failed_instances
+    failed_rows_count = failed_rows.length
+    valid_rows = total_rows_count - failed_rows_count
+    validation_warnings = failed_rows.sort { |a, b| a.errors[:row].first.to_i <=> b.errors[:row].first.to_i }
+                                     .map(&:display_errors_with_row)
```

Comment:
I'm having a hard time deciphering what is happening here. Could this possibly be broken up somehow for readability?

One obvious thing that trips me up, I'm not sure what `|a, b|` each represents... is `failed_rows` a Hash or Array?